ha/core/support_bundle/ha_support_bundle.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import sys
+import os
+import shutil
+import tarfile
+import argparse
+
+from ha.core.error import SupportBundleError
+from ha.const import _DELIM, _sb_default_path, _sb_tar_name, _sb_tmp_src
+from cortx.utils.const import CLUSTER_CONF_LOG_KEY
+from cortx.utils.conf_store import Conf, MappedConf
+
+
+class HASupportBundle:
+
+    @staticmethod
+    def generate(bundle_id: str, target_path: str, cluster_conf_url: str, **filters):
+        """Generate a tar file."""
+        # duration = filters.get('duration', 'P5D')
+        # size_limit = filters.get('size_limit', '500MB')
+        # binlogs = filters.get('binlogs', False)
+        # coredumps = filters.get('coredumps', False)
+        # stacktrace = filters.get('stacktrace', False)
+        # TODO process duration, size_limit, binlogs, coredumps and stacktrace
+        # Find log dirs
+        cluster_conf = MappedConf(cluster_conf_url)
+        log_base = cluster_conf.get(CLUSTER_CONF_LOG_KEY)
+        local_base = cluster_conf.get(
+            f'cortx{_DELIM}common{_DELIM}storage{_DELIM}local')
+        machine_id = Conf.machine_id
```

Comment:
Does Conf need to be loaded here ? or will be loaded in all cases ?

Reply:
Not needed.
`
[root@cortx-ha-headless-svc SB123TEST]# python3
Python 3.6.8 (default, Oct 13 2020, 16:18:22)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from cortx.utils.conf_store import Conf
>>>
>>> Conf.machine_id
'6801f7922f244750be6c7cea6db39ab5'
`